From 01e82ed3e23ae9f47ab026d151cda63e8733971f Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 05 Jan 2021 19:18:40 +0200
Subject: [PATCH] [SW-27979] i2c indirect: status contains rsdl

Accroding to i2c indirect protocol, when state is TRANSMIT status needs
to contain RSDL from RDATA.
When starting a new transaction, RSDL is zeroed out.

Change-Id: I37f8f7f94106bccb26e2a5968b77017237e1dbb3
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index 5975023..c12e6f1 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -345,6 +345,7 @@
 static inline void i2csi_req_reset(void)
 {
 	i2csi_req.sts.err = HL_I2CI_REQ_ERR_NONE;
+	i2csi_req.sts.rsdl = 0;
 	i2csi_req.chunk.hdr.rqcl = 0;
 	i2csi_req.rd.data = NULL;
 	i2csi_req.rd.length = 0;
@@ -424,6 +425,7 @@
 				i2csi_req.rd.length = i2csi_req.wd.hdr.rqdl;
 
 			i2csi_req.rd.hdr.rsdl = i2csi_req.wd.hdr.rqdl;
+			i2csi_req.sts.rsdl = i2csi_req.wd.hdr.rqdl;
 			i2csi_req.sts.sm_sts = HL_I2CI_REQ_ST_TRANSMIT;
 		} else {
 			i2csi_req.sts.sm_sts = HL_I2CI_REQ_ST_IDLE;
@@ -483,6 +485,7 @@
 
 	i2csi_req.chunk.offset = 0;
 	i2csi_req.sts.sm_sts = HL_I2CI_REQ_ST_IDLE;
+	i2csi_req.sts.rsdl = 0;
 	i2csi_req.sts.err = HL_I2CI_REQ_ERR_P_DATA_SIZE;
 }
 
@@ -522,5 +525,6 @@
 void i2csi_req_set_idle(void)
 {
 	i2csi_req.sts.sm_sts = HL_I2CI_REQ_ST_IDLE;
+	i2csi_req.sts.rsdl = 0;
 	i2csi_req.sts.err = HL_I2CI_REQ_ERR_NONE;
 }
